### ISSUE TITLE
XSS Vulnerability 

### ISSUE THREAD

**Issue by Starmann** [MediaBrowser/Emby#3579]
Emby allows you to run arbitrary scripts in the HTML using the video tag. Try editing the Metadata and adding the following to any field that accepts arbitrary strings

```html
<video><source onerror=eval(atob(this.id)) id=YWxlcnQoJ09tYWUgd2EgbW91IFNoaW5kZWlydScp>
```

For bonus fun, this is data that can be downloaded from the internet

**Comment by LukePulverenti** [comment]
Hi, what fields have you tested this with?

**Comment by nomandera** [comment]
https://en.wikipedia.org/wiki/Responsible_disclosure

tl;dr In future

` responsible disclosure is a vulnerability disclosure model in which a vulnerability or an issue is disclosed only after a period of time that allows for the vulnerability or issue to be patched or mended.`

**Comment by Starmann** [comment]
@anoma My apologies. Is there a particular place where I can easily report such vulnerabilities to Emby in the future?

@LukePulverenti I've tested it with the description field. I believe the title field also accepts arbitrary HTML (I was able to make the title be in italics)

**Comment by LukePulverenti** [comment]
How did you get the onerror into the video element? did you edit an html file?

**Comment by LukePulverenti** [comment]
> @anoma My apologies. Is there a particular place where I can easily report such vulnerabilities to Emby in the future?

You can email us at apps@emby.media or send me a PM in our forums.

Regarding this issue though, while we will of course fix it, you have to have admin rights on the server to even get to the metadata editor.

**Comment by Starmann** [comment]
@LukePulverenti It's not the video element that actually plays the regular video content. It's a dummy video element that you can embed in the description. Make sure that the metadata that gets downloaded from the internet goes through the same checks before being loaded by Emby, as that's the attack surface that's more worrying.

Originally, there was a much smaller issue that prompted me to test this. The Anime metadata plugin was creating values that the program wouldn't allow under normal conditions. It was creating 2 decimal places for ratings instead of 1, which prompted me to  poke around and figure out if there was a vulnerability related to input data. So I tried doing some boilerplate from XSShunter just to see. Although it blocked the script tag, events were still usuable.

The best solution would be to have a whitelist of HTML tags, and not allow event attributes.

**Comment by LukePulverenti** [comment]
Well the quick fix for now is to encode them for display. Unfortunately that will upset some people who are embedding markup into descriptions, but at  least that will solve the problem and then we can look at allowing certain things from there.

**Comment by LukePulverenti** [comment]
This is resolved in Emby Server 4.3.1. If you find any other examples, please let me know. Thanks.
